extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/exec/MSQFaultsTest.java

```diff
@@ -374,4 +378,47 @@ public void testTopLevelUnionAllWithJoins()
                 "SQL requires union between inputs that are not simple table scans and involve a filter or aliasing"))
         .verifyPlanningErrors();
   }
+
+  @Test
+  public void testInsertWithReplaceAndExcludeLocks()
+  {
+    for (TaskLockType taskLockType : new TaskLockType[]{TaskLockType.EXCLUSIVE, TaskLockType.REPLACE}) {
+      testLockTypes(
+          taskLockType,
+          "insert into foo1 select * from foo partitioned by day",
```

Comment:
nit: Better capitalization in the test cases. 
```suggestion
          "INSERT INTO foo1 SELECT * FROM foo PARTITIONED BY DAY",
```